src/lib/realmdb/RealmDB.js

```diff
@@ -407,6 +412,27 @@ class RealmDB implements DB, ProfileDB {
   async deleteProfile(): Promise<boolean> {
     return this.profiles.deleteOne({ user_id: this.user.id })
   }
+
+  /**
+   *
+   * @param {TransactionData} data
+   * @returns {Promise<any>}
+   */
+  // eslint-disable-next-line require-await
+  async addToOutbox(data: TransactionData): Promise<any> {
+    return this.inboxes.insertOne({ user_id: this.user.id, ...data })
+  }
+
+  /**
+   *
+   * @param {string} txHash
+   * @returns {Promise<TransactionDetails>}
+   */
+  async getFromOutbox(txHash: string): Promise<TransactionDetails> {
```

Comment:
add recipientPublicKey, perform find by txHash + recipientPublicKey
i understand txHash should be enought by we need to keep the interface defined.
in this case - the outbox key (tx + recipient)

Reply:
async getFromOutbox(recipientPublicKey: string, txHash: string)
RealmDB should incapsulate **RAW DB Queries**
the more high level stuff should be in user storage
feed storage should read privateKey.public.toString()and pass it to feedDB.getFromOutbox()